src/main/java/org/wso2/carbon/connector/utils/MessageBuilder.java

```diff
@@ -295,6 +315,43 @@ private void addAttachment(MimeMultipart multipart, String filePath) throws Mess
         }
     }
 
+    /**
+     * Get content from request body and add the attachment
+     *
+     * @param multipart Multi part body the messages should be added to
+     * @param attachment  the attachment
+     * @throws MessagingException if failed to set attachments
+     * @throws IOException        if an error occurred while reading attachment content
+     */
+    private void addAttachment(MimeMultipart multipart, JSONObject attachment) throws MessagingException, IOException {
+
+        String attachmentName;
```

Comment:
shall we move this definition also inside try

Reply:
Done